### PR TITLE
feat: support provider endpoint in jina executor

### PR DESCRIPTION
**Pull request by zac-li**
## Goals ##

Added `--provider-endpoint` in `jina executor serve` so that we can specify the endpoint used in SageMaker integration for `/invocations`. 

This is needed for Colbert use case as in the Executor we have multiple endpoints, and each SageMaker model package needs to use different endpoint for `/invocations` from the same Executor container.

### CODE REVIEW

File: jina/enums.py [hunk 1]
```diff
@@ -271,6 +271,14 @@ class ProviderType(BetterEnum):
     SAGEMAKER = 1  #: AWS SageMaker
 
 
+class ProviderEndpointType(BetterEnum):
+    """Provider endpoint type."""
+
+    NONE = 0
+    RANK = 1
+    ENCODE = 2
```
Comment (deepankarm) [inline jina/enums.py:279]: Any reason we have rank & encode hard-coded here?

Comment (JoanFM) [reply jina/enums.py:279]: oh yes, this should not be an enum. This should be free

Comment (zac-li) [reply jina/enums.py:279]: ok I can remove that

Comment (JoanFM) [reply jina/enums.py:279]: pls remove yes

File: jina/orchestrate/deployments/__init__.py [hunk 2]
```diff
@@ -35,6 +35,7 @@
     PollingType,
     ProtocolType,
     ProviderType,
+    ProviderEndpointType,
```
Comment (JoanFM) [inline jina/orchestrate/deployments/__init__.py:38]: remove

File: jina/parsers/orchestrate/pod.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 from dataclasses import dataclass
 from typing import Dict
 
-from jina.enums import PodRoleType, ProtocolType, ProviderType
+from jina.enums import PodRoleType, ProtocolType, ProviderEndpointType, ProviderType
```
Comment (JoanFM) [inline jina/parsers/orchestrate/pod.py:7]: remove import

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Review by JoanFM — APPROVED** [review]

**Pull request merged**